apps/api/src/api.service.ts

```diff
@@ -28,6 +51,97 @@ export class ApiService {
     };
   }
 
+  async validateAssets(content: AssetIncludedRequestDto): Promise<void> {
+    const checkingList: Array<{ onlyImage: boolean; referenceId: string }> = [];
+    if (content.profile) {
+      content.profile.icon?.forEach((reference) => checkingList.push({ onlyImage: true, referenceId: reference.referenceId }));
+    } else if (content.content) {
+      content.content.assets?.forEach(
+        (asset) =>
+          asset.references?.forEach((reference) =>
+            checkingList.push({
+              onlyImage: false,
+              referenceId: reference.referenceId,
+            }),
+          ),
+      );
+    }
+
+    const redisResults = await Promise.all(checkingList.map((obj) => this.redis.get(getAssetMetadataKey(obj.referenceId))));
+    const errors: string[] = [];
+    redisResults.forEach((res, index) => {
+      if (res === null) {
+        errors.push(`${content.profile ? 'profile.icon' : 'content.assets'}.referenceId ${checkingList[index].referenceId} does not exist!`);
+      } else if (checkingList[index].onlyImage) {
+        // checks if attached asset is an image
+        const metadata: IAssetMetadata = JSON.parse(res);
+        if (!isImage(metadata.mimeType)) {
+          errors.push(`profile.icon.referenceId ${checkingList[index].referenceId} is not an image!`);
+        }
+      }
+    });
+    if (errors.length > 0) {
+      throw new HttpErrorByCode[400](errors);
+    }
+  }
+
+  // TODO: make all these operations transactional
```

Comment:
jobs have limited size of 500MB so it was not feasible to store the content directly in the job. Had to store it separately and all of these should be changed to be transactional